client/state/audio/middleware.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_RECEIVE_EVENT,
+} from 'state/action-types';
+
+const isAudioSupported = () => typeof window === 'object' && typeof window.Audio === 'function';
```

Comment:
of note, we've had plenty of discussion on creating a "browser object model" in Calypso which we could use for all browser-related functionality and still provide for easier testing and mocking. just a thought.